### PR TITLE
util/filesort: fix data race

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

File: util/filesort/filesort.go [hunk 1]
```diff
@@ -560,11 +557,6 @@ func (w *Worker) input(row *comparableRow) error {
 		w.ctx.external = true
 		go w.flushToFile()
 	}
-
-	if w.err != nil {
```
Comment (shenli) [inline util/filesort/filesort.go]: Why not check err here?

Comment (coocood) [inline util/filesort/filesort.go]: the error is set on the background, we don't need to check it here.
It will be checked later when Output is called.

Comment (shenli) [inline util/filesort/filesort.go]: It may be overwrited in flushToFile().

Comment (shenli) [inline util/filesort/filesort.go]: And in which line the error is checked?

Comment (coocood) [inline util/filesort/filesort.go]: https://github.com/pingcap/tidb/blob/26ebc019f8b5dd564244dadb6f1ea0b5b60b96e9/util/filesort/filesort.go#L331

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**